### ISSUE TITLE
Is it possible to add GPU running

### ISSUE THREAD

**Issue by Container-Zero** [corkborg/wd14-tagger-standalone#2]
First of all thank you for your work, this project is very helpful for me, but could you please make it easy to add a way for it to run on `GPU`? Currently `CPU` running can easily trigger `CPU` bottlenecks, looking forward to your reply!

**Comment by corkborg** [comment]
@Container-Zero 
Hi

I previously tried to run it on a GPU but gave up.
The reason is that the versions of CUDA and cuDNN required for onnixruntime==1.16 are too old.

The recently released onnixruntime==1.17 supports newer versions, so it might be worth a try.

**Comment by corkborg** [comment]
I tried it and was able to run it on a GPU.

PyPI version of onnxruntime-gpu does not support CUDA 12.2, so it seems you need to use Azure Devops version.

Inference on GPU was very fast (except for mldanbooru type models).

https://github.com/corkborg/wd14-tagger-standalone/pull/4

**Comment by Container-Zero** [comment]
> I tried it and was able to run it on a GPU.
> 
> PyPI version of onnxruntime-gpu does not support CUDA 12.2, so it seems you need to use Azure Devops version.
> 
> Inference on GPU was very fast (except for mldanbooru type models).
> 
> #4

Thank you very much for your efforts, I tried it and your project worked great! But due to time urgency, I wrote similar code implementing the same multi-threaded `GPU` computing functionality about a week ago, and at that time I was in a slightly different situation than you, as I was running in a `Conda` environment, and the highest version supported by `CudaToolKit` was only up to `1.18`, so instead I needed to downgrade onnxruntime-gpu downgrade to `1.15.0` and use the old `CUDNN`. All in all, it all worked out as expected. Thanks again for all your efforts. Cheers!
